lib/HttpClient/CurlClient.php

```diff
@@ -218,17 +229,24 @@ private function executeRequestWithRetries($opts, $absUrl)
             $rcode = 0;
             $errno = 0;
 
-            $curl = curl_init();
-            curl_setopt_array($curl, $opts);
-            $rbody = curl_exec($curl);
+            if (!is_null($this->curlHandle) && Stripe::getEnablePersistentConnections()) {
+                curl_reset($this->curlHandle);
+            } else {
+                $this->curlHandle = curl_init();
```

Comment:
Shouldn't we close the old handle before creating a new one? Your first if means that you can have it initialized and not cleaned up (for example with an exception on the previous request) right?

Reply:
👍